crates/host/src/wasmbus/mod.rs

```diff
@@ -2529,10 +2542,11 @@ impl Host {
                             break;
                         }
                     }
-                    if remaining == 0 {
-                        drop(instances);
-                        entry.remove();
-                    }
+                }
+
+                if actor_instances.is_empty() {
```

Comment:
Is this right? Previously we would drop the individual `instances` if a group of annotated-instances became empty. Now it looks like we only drop if _all_ instances are gone for the actor

Though it also looks like there's a bug in the current implementation. I don't think we want to `remove` when _any_ group becomes empty. So it looks like we might need two checks:
- remove groups from `actor_instances` if they become empty as a result of scaling down
- remove the overall actor entry from `actors` if all instances are stopped

Reply:
I thought the behavior you described is actually what I implemented, but will take another look back through because that's definitely desired behavior.

Reply:
Fairly sure my latest commit now implements this behavior as described